packages/nodes-base/nodes/Html/Html.node.ts

```diff
@@ -0,0 +1,362 @@
+					}
+
+					returnData.push(...makeExecutionData.call(this, html, itemIndex));
+				} else if (operation === 'extractHtmlContent') {
+					// ----------------------------------
+					//         extractHtmlContent
+					// ----------------------------------
+
+					const dataPropertyName = this.getNodeParameter('dataPropertyName', itemIndex);
+					const extractionValues = this.getNodeParameter(
+						'extractionValues',
+						itemIndex,
+					) as IDataObject;
+					const options = this.getNodeParameter('options', itemIndex, {});
+					const sourceData = this.getNodeParameter('sourceData', itemIndex) as string;
+
+					item = items[itemIndex];
+
+					let htmlArray: string[] | string = [];
+					if (sourceData === 'json') {
+						if (item.json[dataPropertyName] === undefined) {
+							throw new NodeOperationError(
+								this.getNode(),
+								`No property named "${dataPropertyName}" exists!`,
+								{ itemIndex },
+							);
+						}
+						htmlArray = item.json[dataPropertyName] as string;
+					} else {
+						if (item.binary === undefined) {
+							throw new NodeOperationError(
+								this.getNode(),
+								'No item does not contain binary data!',
+								{
+									itemIndex,
+								},
+							);
+						}
+						if (item.binary[dataPropertyName] === undefined) {
+							throw new NodeOperationError(
+								this.getNode(),
+								`No property named "${dataPropertyName}" exists!`,
+								{ itemIndex },
+							);
+						}
+
+						const binaryDataBuffer = await this.helpers.getBinaryDataBuffer(
+							itemIndex,
+							dataPropertyName,
+						);
+						htmlArray = binaryDataBuffer.toString('utf-8');
+					}
+
+					// Convert it always to array that it works with a string or an array of strings
+					if (!Array.isArray(htmlArray)) {
+						htmlArray = [htmlArray];
+					}
+
+					for (const html of htmlArray as string[]) {
+						const $ = cheerio.load(html);
+
+						const newItem: INodeExecutionData = {
+							json: {},
+							pairedItem: {
+								item: itemIndex,
+							},
+						};
+
+						// Itterate over all the defined values which should be extracted
+						let htmlElement;
+						for (const valueData of extractionValues.values as IValueData[]) {
+							htmlElement = $(valueData.cssSelector);
+
+							if (valueData.returnArray) {
+								// An array should be returned so itterate over one
+								// value at a time
+								newItem.json[valueData.key] = [];
+								htmlElement.each((i, el) => {
+									(newItem.json[valueData.key] as Array<string | undefined>).push(
+										getValue($(el), valueData, options),
+									);
+								});
+							} else {
+								// One single value should be returned
+								newItem.json[valueData.key] = getValue(htmlElement, valueData, options);
+							}
+						}
+						returnData.push(newItem);
+					}
+				}
+			} catch (error) {
+				if (this.continueOnFail()) {
+					returnData.push(...makeExecutionData.call(this, error.message, itemIndex));
```

Comment:
If continue on fail the call to makeExecutionData will return `[{ html: "error message" }]` instead of `[{ error: "error message" }]`, which makes it hard to distiguish from valid results.

Reply:
Restored the original implementation for `continueOnFail`.